File: web/src/lib/lq-ai/api/intakeBridges.ts
/**
 * Admin intake-bridges API client — M3-D4.
 *
 * Surface:
 *
 *   - GET    /api/v1/admin/intake-bridges           — list live installs
 *   - DELETE /api/v1/admin/intake-bridges/slack/{id} — soft-delete
 *   - DELETE /api/v1/admin/intake-bridges/teams/{id} — soft-delete
 *
 * All endpoints are admin-gated server-side; non-admin users get a
 * 403 which `apiRequest` surfaces as a typed `ForbiddenError` the
 * admin page catches and renders inline.
 */
import { apiRequest } from './client';

export interface SlackWorkspaceSummary {
	id: string;
	team_id: string;
	team_name: string;
	installer_slack_user_id: string;
	installed_at: string;
}

export interface TeamsTenantSummary {
	id: string;
	tenant_id: string;
	tenant_name: string;
	installer_oid: string;
	installed_at: string;
}

export interface IntakeBridgesList {
	slack_workspaces: SlackWorkspaceSummary[];
	teams_tenants: TeamsTenantSummary[];
}

export async function listIntakeBridges(): Promise<IntakeBridgesList> {
	return apiRequest<IntakeBridgesList>('/admin/intake-bridges', { method: 'GET' });
}

export async function deleteSlackWorkspace(workspaceId: string): Promise<void> {
	await apiRequest<void>(`/admin/intake-bridges/slack/${workspaceId}`, {
		method: 'DELETE'
	});
}

export async function deleteTeamsTenant(tenantId: string): Promise<void> {
	await apiRequest<void>(`/admin/intake-bridges/teams/${tenantId}`, {
		method: 'DELETE'
	});
}

File: api/app/schemas/intake_bridges.py
"""Pydantic wire shapes for the admin intake-bridges surface — M3-D4.

Used by ``GET /api/v1/admin/intake-bridges`` to surface what's
currently connected in the LQ.AI admin UI. The shapes are deliberately
section-split (slack_workspaces + teams_tenants) rather than a single
polymorphic list because the UI renders two distinct sections
(per [M3-D4 plan](../../docs/M3-IMPLEMENTATION-PLAN.md#task-m3-d4--bot-configuration-in-lqai-admin-ui))
and a section-split shape lets the client render directly without
discriminator-key gymnastics.
"""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class SlackWorkspaceSummary(BaseModel):
    """One live Slack workspace install for the admin list."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    team_id: str
    team_name: str
    installer_slack_user_id: str
    installed_at: datetime


class TeamsTenantSummary(BaseModel):
    """One live Teams tenant install for the admin list."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    tenant_id: str
    tenant_name: str
    installer_oid: str
    installed_at: datetime


class IntakeBridgesList(BaseModel):
    """Section-split response for the admin intake-bridges list."""

    slack_workspaces: list[SlackWorkspaceSummary]
    teams_tenants: list[TeamsTenantSummary]
